### PR TITLE
MQE-1306: Can not get actual entity data

### PR DESCRIPTION
**Pull request by tomreece**
### Description
Fixes bug [MQE-1306](https://jira.corp.magento.com/browse/MQE-1306) by adding a fallback checking for `camelCaseKeys` in data if now `lowercasekeys` are found.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/verification tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)
 - [ ] Changes to Framework doesn't have backward incompatible changes for tests or have related Pull Request with fixes to tests

### CODE REVIEW

File: src/Magento/FunctionalTestingFramework/DataGenerator/Objects/EntityDataObject.php [hunk 1]
```diff
@@ -177,15 +177,22 @@ public function getDataByName($name, $uniquenessFormat)
 
         $name_lower = strtolower($name);
 
-        if ($this->data !== null && array_key_exists($name_lower, $this->data)) {
+        if ($this->data === null) {
+            return null;
+        }
+
+        if (array_key_exists($name_lower, $this->data)) {
```
Comment (KevinBKozan) [inline src/Magento/FunctionalTestingFramework/DataGenerator/Objects/EntityDataObject.php]: `$this->data` can still be camelCased, so you need to compare `lower_name` === `lower_index`.
Could leverage `array_change_key_case()` for a temp array if needed

Comment (tomreece) [reply src/Magento/FunctionalTestingFramework/DataGenerator/Objects/EntityDataObject.php]: I don't quite understand.

The second branch of this conditional `elseif (array_key_exists($name, $this->data))` catches the cases where the key is camelCased.

Comment (KevinBKozan) [reply src/Magento/FunctionalTestingFramework/DataGenerator/Objects/EntityDataObject.php]: Honestly I'm not sure anymore either. I was running the attached usecase to the ticket and it was causing mis-matching issues.
Reran to get some proof and it's no longer failing. So disregard this comment.

Comment (tomreece) [reply src/Magento/FunctionalTestingFramework/DataGenerator/Objects/EntityDataObject.php]: Haha cool. I was able to run the usecase attached to the ticket successfully and with my included unit test I think we're safe.

**Pull request merged**